
## How to read the unified table	
Each row correspond to a variant in a locus that have been associated to AD GWAS (Combining finemaping and Coloc results for the 4 GWAS studies data used). 	

For each AD locus, only the variants with inclusion score>0.1 are included or, if no variant with inclusion score>0.1, only the leading variants in term of inclusion score for GWAS or for an xQTL are displayed.  
Inclusion score correspond to the maximal value for its associated PIPs or VCP (Variant coloc probability) across any GWAS (for GWAS part), or xQTL (for xQTL part)

If the variant have been associated to an xQTL, summary are shown in xQTLs summary part.	 

Contexts are ordered and green colored according to a confidence score (C1 / darkgreen is best, C6/light green is lowest confidence) which combined TWAS, MR, finemapping and coloc results:	 

- C1: cTWAS/MR, cs95 single context fine-mapping overlap
- C2: cTWAS/MR, colocalization
- C3: TWAS and (cs95 fine-mapping overlap OR colocalization)
- C4: cs95 single context fine-mapping overlap
- C5: colocalization OR any fine-mapping overlap (multicontext, cs50, cs70..)
- C6: TWAS only

The '+' or '-' after the context name correspond to the direction of the effect on the gene/molecular trait observed in finemapping (first '+/-') or in colocboost (second '+/-'). 
'Effect' column correspond to the effect on the molecular traits, 
  



For more details please check out [this page](https://statfungen.github.io/xqtl-resources/xqtl_resource_description/#alzheimers-disease-gwas-integration)

- `unified_AD_loci_xQTL_summary.xlsx` created by Alexandre Pelletier. Latest version from Aug 16, 2025 containing 197 genes
- `context_meta.tsv` generated by Ru Feng
```bash
cat /mnt/vast/hpc/homes/rf2872/codes/xqtl-analysis/analysis/Wang_Columbia/susie_twas/*/commands_to_submit.txt |grep -oP '(?<=--phenotype-names )[^\\]*'| sed 's/\s--.*$//' | awk -v OFS=',' '{for (i=1; i<=NF; i++) printf "%s%s", $i, (i==NF?ORS:OFS)}'   > context.txt
cat /mnt/vast/hpc/homes/rf2872/codes/xqtl-analysis/analysis/Wang_Columbia/susie_twas/*/commands_to_submit.txt |grep -oP '(?<=--name )[^\\]*'| sed 's/\s--.*$//' |sed 's/ //g' > analysis_name.txt
 

paste context.txt analysis_name.txt | awk '{printf "%-30s %s\n", $1, $2}' > combined.txt


cat /mnt/vast/hpc/homes/rf2872/codes/xqtl-analysis/analysis/Wang_Columbia/fsusie/*/commands_to_submit.txt |grep -oP '(?<=--phenotype-names )[^\\]*'| sed 's/\s--.*$//' | awk -v OFS=',' '{for (i=1; i<=NF; i++) printf "%s%s", $i, (i==NF?ORS:OFS)}'   > context.txt
cat /mnt/vast/hpc/homes/rf2872/codes/xqtl-analysis/analysis/Wang_Columbia/fsusie/*/commands_to_submit.txt |grep -oP '(?<=--name )[^\\]*'| sed 's/\s--.*$//' |sed 's/ //g' > analysis_name.txt
 

paste context.txt analysis_name.txt | awk '{printf "%-30s %s\n", $1, $2}' >> combined.txt



cat /mnt/vast/hpc/homes/rf2872/codes/xqtl-analysis/analysis/Wang_Columbia/susie_twas_trans/*/commands_to_submit.txt |grep -oP '(?<=--phenotype-names )[^\\]*'| sed 's/\s--.*$//' | awk -v OFS=',' '{for (i=1; i<=NF; i++) printf "%s%s", $i, (i==NF?ORS:OFS)}'   > context.txt
cat /mnt/vast/hpc/homes/rf2872/codes/xqtl-analysis/analysis/Wang_Columbia/susie_twas_trans/*/commands_to_submit.txt |grep -oP '(?<=--name )[^\\]*'| sed 's/\s--.*$//' |sed 's/ //g' > analysis_name.txt
 

paste context.txt analysis_name.txt | awk '{printf "%-30s %s\n", $1, $2}' >> combined.txt
```


```R
df <- fread('/mnt/vast/hpc/homes/rf2872/codes/CUMC_fungun_xqtl_analysis/analysis/Others/combined.txt', sep = ' ', header = FALSE)
colnames(df) <- c('context', 'analysis_name')

df <- df %>% mutate(cohort = str_split(analysis_name, "_", simplify = T) %>% .[,1])  %>% filter(!str_detect(analysis_name, 'leafcutter2'))
df <- df[,c('cohort','context','analysis_name')]
write_delim(df, '/mnt/vast/hpc/homes/rf2872/codes/xqtl-analysis/resource/context_meta.tsv', delim = '\t')
```


